### PR TITLE
Add Fail2ban support

### PR DESCRIPTION
**Pull request by lapineige**
## Problem
- Wallabag is not protected by fail2ban against brute force
See #37 

## Solution
- Add fail2ban config.

The number of retry is set to 5, I think it allows users to make a reasonable number of errors while restricting brute force possibilities.

**Warning** : this PR will drop support for version older than 3.5, in particular Yunohost 2.7 (Debian 8).

## PR Status
- [x] Code finished.
- [ ] Tested with Package_check.
- [x] Fix or enhancement tested.
- [ ] Upgrade from last version tested.
- [x] Can be reviewed and tested.

## Validation
---
*Minor decision*
- **Upgrade previous version** : 
- [x] **Code review** : Maniack C
- [x] **Approval (LGTM)** : Maniack C
- [x] **Approval (LGTM)** : JimboJoe
- **CI succeeded** : 
[Build Status](https://ci-apps-hq.yunohost.org/jenkins/job/wallabag2_ynh%20PR65/)  
When the PR is marked as ready to merge, you have to wait for 3 days before really merging it.

### CODE REVIEW

**Comment by lapineige** [comment]
Changed the base branch to master, as it is an high priority issue, we can't wait for testing to be merged.

**Comment by lapineige** [comment]
I fixed the regex conf, at first I did not understand how to configure it correctly.

I tried a fresh install + removal, it works. Fail2ban seems to be active: after 5 failed login attempts, the login page doesn't load.

It's ready for review.

File: scripts/remove [hunk 1]
```diff
@@ -55,8 +55,24 @@ ynh_remove_nginx_config
...
+
+#=================================================
+
+# REMOVE DEDICATED USER
+
+#=================================================
+
+
+
+
```
Comment (maniackcrudelis) [inline scripts/remove]: To be removed

File: scripts/restore [hunk 2]
```diff
@@ -76,12 +76,19 @@ ynh_system_user_create $app	# Recreate the dedicated user, if not existing
 
 chown -R $app: $final_path
 
+#=================================================
+# RESTORE FAIL2BAN CONFIGURATION
+#=================================================
+ynh_restore_file "/etc/fail2ban/jail.d/$app.conf"
+ynh_restore_file "/etc/fail2ban/filter.d/$app.conf"
+
+systemctl reload fail2ban # Reload instead of restart for better performance
```
Comment (maniackcrudelis) [inline scripts/restore]: Use `ynh_systemd_action` instead of systemctl
https://yunohost.org/#/packaging_apps_helpers_en

**Comment by maniackcrudelis** [comment]
Adding fail2ban is a good improvement, not a security issue needed to be quickly fixed.

File: scripts/remove (hunk 1 above)
Comment (lapineige) [reply scripts/remove]: Why ?

File: scripts/restore (hunk 2 above)
Comment (lapineige) [reply scripts/restore]: Done :)

**Comment by lapineige** [comment]
Users password are not protected right now, and it's the same as the SSO password, so it would be a lot better to prevent brute force attack. It's not a security flaw, but still pretty important.

File: scripts/remove (hunk 1 above)
Comment (maniackcrudelis) [reply scripts/remove]: Because that's a duplicated comment with blank lines.

**Comment by maniackcrudelis** [comment]
User passwords are protected by https.
Brute force attack aren't prevented indeed and that's a pretty good improvement. Not a security issue.

File: scripts/remove (hunk 1 above)
Comment (lapineige) [reply scripts/remove]: Oh ok, in my mail it was shown differently, and `ynh_remove_fail2ban_config` was included.
I'll do it right now.

**Comment by JimboJoe** [comment]
The CI job is failing... :thinking:

**Comment by lapineige** [comment]
Do you know why ? (where can I find the logs ?)

**Comment by JimboJoe** [comment]
Yep, you click on the build badge, then "last build", then "console output".

**Comment by lapineige** [comment]
This page ? https://yunohost.github.io/ci/

**Comment by maniackcrudelis** [comment]
There https://ci-apps-hq.yunohost.org/jenkins/job/wallabag2_ynh%20PR65/5/console

**Comment by lapineige** [comment]
Ok, only upgrade from this version fails…

I found the issue: I fixed the regex syntax in the install script, but not in the upgrade one
Should be fixed now.

**Comment by JimboJoe** [comment]
Upgrade is stille [failing on the CI](https://ci-apps-hq.yunohost.org/jenkins/job/wallabag2_ynh%20PR65/lastBuild/console).
```
Warning: yunohost.hook <lambda> - [3344.1] Job for fail2ban.service failed because the control process exited with error code.
Warning: yunohost.hook <lambda> - [3344.1] See "systemctl status fail2ban.service" and "journalctl -xe" for details.
Warning: yunohost.hook <lambda> - [3344.1] !!
Warning: yunohost.hook <lambda> - [3344.1]   wallabag2's script has encountered an error. Its execution was cancelled.
Warning: yunohost.hook <lambda> - [3344.1] !!
Warning: yunohost.hook <lambda> - [3344.1] ./upgrade: line 61: ynh_backup_after_failed_upgrade: command not found
```

**Comment by lapineige** [comment]
But we don't have the log for fail2ban… :(

**Comment by JimboJoe** [comment]
Isn't the problem reproducible with package_check in your environment?

**Comment by lapineige** [comment]
How do I use it ?

**Comment by JimboJoe** [comment]
As an app maintainer, you will love [package_check](https://github.com/YunoHost/package_check)! Have a look at the README, you can install it on your test server/VM. It's been developed mainly by @maniackcrudelis, and it's what produces the results of the CI.

**Comment by lapineige** [comment]
I don't have any VM or test server for that :/

**Comment by JimboJoe** [comment]
Then you'll be interested by [this forum post](https://forum.yunohost.org/t/proposition-for-a-new-non-official-ci/3595/6?u=jimbojoe) :wink:

**Comment by lapineige** [comment]
Ok, I tried again on my main server (backup + remove old wallabag + install master + upgrade to this branch), the error is:
> No file(s) found for glob /var/www/wallabag2/var/logs/prod.log

I fixed that in the install script, but not in the upgrade script :sweat_smile:
I'll do it.

**Comment by JimboJoe** [comment]
By the way, more relevant link [here](https://github.com/YunoHost/doc/blob/master/packaging_apps_ci.md) to use the CI available for app packagers :wink:

**Comment by lapineige** [comment]
CI is succeeding right now.

Let's merge ?

**Comment by JimboJoe** [comment]
Can be merged in 3 days (if @maniackcrudelis confirms his code review after late changes)

**Comment by JimboJoe** [comment]
Can be merged in 3 days.

**Comment by lapineige** [comment]
Thanks @maniackcrudelis for the improvements :)

**Pull request merged**